### PR TITLE
Fixed Lint issues and SNS public issue

### PR DESCRIPTION
**Pull request by troy-ameigh**
1. Fixed Minor Taskcat Lint Errors:
     -- elasticsearch.yaml - added "Resource: "arn:aws🇪🇸AWS-REGION:ACCOUNT-id:*""
     -- glue.yaml - removed unused parameter: "CuratedBucketName"
     -- iam-roles.yaml - removed unused parameter: "AthenaQueryResultsBucketARN"
     -- data-lake.yaml - removed unused parameters: "AvailabilityZones" "KeyPairName"         "NotebookTrainingInstanceType"
All parameters defined and used in other template files.

2. Fixed Public SNS issue
   -- buckets.yaml - In SNS topic policy changed Principal:
           AWS: !Ref 'AWS::AccountId'

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: templates/buckets.yaml [hunk 1]
```diff
@@ -1,11 +1,11 @@
-AWSTemplateFormatVersion: "2010-09-09"
+AWSTemp
+lateFormatVersion: "2010-09-09"
```
Comment (sshvans) [inline templates/buckets.yaml]: Shouldn't this be in single line?

File: templates/data-lake.yaml [hunk 2]
```diff
@@ -118,11 +115,6 @@ Conditions:
     - !Ref 'EnableRedshift'
     - 'yes'
 Parameters:
-  AvailabilityZones:
```
Comment (sshvans) [inline templates/data-lake.yaml]: Is this parameter not being used?

File: templates/data-lake.yaml [hunk 3]
```diff
@@ -157,10 +149,10 @@ Parameters:
     Description: "Specifies whether Amazon Redshift will be publicly accessible.\
       \ Only applies if EnableRedshift is set to yes."
     Type: String
-  KeyPairName:
```
Comment (sshvans) [inline templates/data-lake.yaml]: Is this parameter not being used?

File: templates/elasticsearch.yaml [hunk 4]
```diff
@@ -132,6 +132,7 @@ Resources:
                   - !Ref "NAT2ElasticIP"
             Effect: Allow
             Principal: "*"
+            Resource: "arn:aws:es:AWS-REGION:ACCOUNT-id:*"
```
Comment (sshvans) [inline templates/elasticsearch.yaml]: I don't think this is the right ARN format

File: templates/buckets.yaml [hunk 5]
```diff
@@ -56,7 +56,7 @@ Resources:
           - Sid: StatementId
             Effect: Allow
             Principal:
-              AWS: "*"
+              AWS: !Ref 'AWS::AccountId'
```
Comment (sshvans) [inline templates/buckets.yaml]: Have you tested, if this is the sufficient for the policy?

File: templates/data-lake.yaml (hunk 3 above)
Comment (troy-ameigh) [reply templates/data-lake.yaml]: correct this parameter is not used. Lint gave and error on it during the taskcat run. and I cannot see in the template where it is used. So I recommended removing it.

File: templates/buckets.yaml (hunk 5 above)
Comment (troy-ameigh) [reply templates/buckets.yaml]: This appears to  be sufficient to remove security alerts for AWS. However I am open to doing something different if there is a better  way.

File: templates/data-lake.yaml (hunk 2 above)
Comment (troy-ameigh) [reply templates/data-lake.yaml]: Correct this parameter is not used. Lint gave and error on it during the taskcat run. and I cannot see in the template where it is used. So I recommended removing it.

File: templates/buckets.yaml (hunk 1 above)
Comment (troy-ameigh) [reply templates/buckets.yaml]: That is correct, not sure how it happened. Already corrected in my repo.

File: templates/elasticsearch.yaml (hunk 4 above)
Comment (troy-ameigh) [reply templates/elasticsearch.yaml]: sshvans, I see what you are saying and want to update it this. 

Resource: "arn:aws:es:${AWS::Region}:${AWS::AccountId}:*"

**Comment by sshvans** [comment]
@troy-ameigh Can you push changes to the PR?

**Comment by troy-ameigh** [comment]
Should be done.


--

Troy Ameigh
Partner Solutions Architect
Amazon Web Services
M: (512)201-0430
Email: ameighta@amazon.com<mailto:ameighta@amazon.com>

[signature_845779016]

[https://www.certmetrics.com/amazon/Telerik.Web.UI.WebResource.axd?imgid=392a4fede6ec438486177f3cb2b1806d&type=rbi]<https://www.certmetrics.com/amazon/public/badge.aspx?i=1&t=c&d=2016-11-14&ci=AWS01065908>[https://www.certmetrics.com/amazon/Telerik.Web.UI.WebResource.axd?imgid=f14e862e42be49ec952fa4990b556bd3&type=rbi]<https://www.certmetrics.com/amazon/public/badge.aspx?i=3&t=c&d=2019-08-15&ci=AWS01065908>[https://www.certmetrics.com/amazon/Telerik.Web.UI.WebResource.axd?imgid=85746a58b26e440e8537abe65142f892&type=rbi]<https://www.certmetrics.com/amazon/public/badge.aspx?i=2&t=c&d=2019-11-27&ci=AWS01065908>

Thoughts on our interaction? Provide feedback here<https://feedback.aws.amazon.com/?ea=ameighta&fn=Troy&ln=Ameigh>.

**Pull request merged**